### DISCUSSION TITLE
Volume rendering of non-tetrahedral UnstructuredGrid

### DISCUSSION THREAD

**Discussion by PProfizi** [pyvista/pyvista#4818] [Q&A]
Hi everyone,

I am trying to use the `add_volume` capability of PyVista on different UnstructuredGrid objects, some having other elements than tetrahedra.

The result is I get this warning, and then no plot, without any actual error being raised:

`vtkUnstructuredGridBuny:930   WARN| vtkUnstructuredGridBunykRayCastFunction (000001EEB10079B0): Input contains more than tetrahedra - only tetrahedra are supported`

It seems then that volume rendering is not supported for UnstructuredGrid objects with elements other than tetrahedron, right?

I just wanted to confirm whether this is the case, or if there is a way to do it properly.

I am asking because the [example on volume rendering](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.add_volume.html) does not state any limitation on the UnstructuredGrid. I only found the limitation stated [here](https://vtk.org/doc/release/5.4/html/a01791.html), but `vtkUnstructuredGridBunykRayCast` si not directly mentioned in the PyVista example, only `vtkUnstructuredGridVolumeRayCast`.

**Comment by MatthewFlamm** [comment]
I believe your analysis looks right, and this information should be added to the documentation including the example section.  Please do so!

> but vtkUnstructuredGridBunykRayCast si not directly mentioned in the PyVista example, only vtkUnstructuredGridVolumeRayCast.

`vtkUnstructuredGridVolumeRayCast` is not mentioned in the documentation as far as I can see, but `vtkUnstructuredGridVolumeRayCastMapper` is.  `vtkUnstructuredGridBunykRayCast` must be used within that mapper.

We may also want to check for this and throw a python warning or error, or do a triangulation.  I am a little hesitant to automatically triangulate as it could easily lead to seg faults on large 3D unstructured grids.
